### ISSUE TITLE
Symbol mangling and package usage

### ISSUE THREAD

**Issue by tagyro** [CocoaPods/cocoapods-packager#113]
I'm developing a pod, let's call it 'MyPod', which has some dependencies, let's say AFNetworking, RNCryptor and CocoaAsyncSocket. 
MyPod in turn is a dependency for another project, My Project, which also has RestKit as a dependency.
This is the final structure:

MyProject
    |---------MyPod
    |           |--------AFNetworking - 2.6.3
    |           |--------RNCryptor
    |           |--------CocoaAsyncSocket
    |---------RestKit
                |--------AFNetworking - 1.3.0

From my understanding, calling 'pod package MyPod.podspec' I should be able to package MyPod and use the resulting folder as the pod source (via :path =>).
Unfortunately this doesn't seem to be possible.
Although I managed to get the 'pod install' working in MyProject, the resulting project is unusable or with missing references (to AFNetworking etc).
Using the -no-mangle option defeats the purpose, as my main goal was to mangle the symbols so that I can have two different versions of AFNetworking in the same project.
Is there something I'm missing? Am I using this wrong?

**Comment by neonichu** [comment]
In theory, this setup should work — you would need to be a bit more specific in order for us to determine what might be going wrong.

**Comment by tagyro** [comment]
@neonichu You're right, in most cases this works. It wasn't working in my case because one of the libraries I was using was still in use in the public header.
I'll open a new issue with a feature request and a sample project to exemplify my somewhat special case.

**Comment by tagyro** [comment]
I've created a new branch of the SDK I'm working on which is available at https://github.com/sensorberg-dev/ios-sdk/tree/v2m 
The SDK has AFNetworking as a dependency. I've managed to move all AFNetworking related calls into "internal" classes which means that the SDK can be "mangled" and another AFNetworking version could be used by a 3rd party.
The problem now is that I'm also using Tolo (an event publish/subscribe framework) and I need that in the header files (but this can be easily solved by including Tolo as a dependency by the 3rd party).
What would be nice at this point is an option to exclude Tolo (or more generally a dependency) from being mangled.
Thank you for your answer (and work) @neonichu and I think this issue could be closed.
